### ISSUE TITLE
Wrong source region when using BLIT_FLIP_X/Y with a partially clipped sprite

### ISSUE THREAD

**Issue by aduros** [aduros/wasm4#11]
Looks like partially offscreen blits with flipping (and maybe also rotation?) are drawn incorrectly.

**Comment by eXponenta** [comment]
Are you sure that bug still is reproduced?
Can i help with it, becuse it a will be required for my game.
My game use a atlass for movement animation, and flips can be used for avoid doubling right-left movements.

**Comment by JerwuQu** [comment]
This is still an issue in 2.2.0. Might give a shot at fixing it later tonight unless it's already in progress...?

```c
#include "wasm4.h"

const uint8_t smiley[] = {
    0b11000011,
    0b10000001,
    0b00100100,
    0b00100100,
    0b00000000,
    0b00100100,
    0b10011001,
    0b11000011,
};

void update () {
    *DRAW_COLORS = 2;

    blit(smiley, 0, 145, 8, 8, BLIT_1BPP);
    blit(smiley, 20, 150, 8, 8, BLIT_1BPP);
    blit(smiley, 40, 155, 8, 8, BLIT_1BPP);
    blit(smiley, 60, 160, 8, 8, BLIT_1BPP);

    blit(smiley, 80, 145, 8, 8, BLIT_1BPP | BLIT_FLIP_Y);
    blit(smiley, 100, 150, 8, 8, BLIT_1BPP | BLIT_FLIP_Y);
    blit(smiley, 120, 155, 8, 8, BLIT_1BPP | BLIT_FLIP_Y);
    blit(smiley, 140, 160, 8, 8, BLIT_1BPP | BLIT_FLIP_Y);
}
```

![image](https://user-images.githubusercontent.com/3710677/149341131-aa4ae680-6677-4771-8b79-0cedc0f194a3.png)

**Comment by aduros** [comment]
Please do @JerwuQu! A fix would be much appreciated :smiley:

**Comment by JerwuQu** [comment]
Fixed in #319
